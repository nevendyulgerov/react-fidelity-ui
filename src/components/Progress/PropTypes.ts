import * as PropTypes from 'prop-types';

export const typeShape = PropTypes.oneOf([
  'primary',
  'secondary',
  'success',
  'info',
  'danger',
  'warning'
]);

export const sizeShape = PropTypes.oneOf([null, 'xs', 'sm', 'md', 'lg', 'xl']);

export const labelShape = PropTypes.oneOfType([
  PropTypes.string,
  PropTypes.number,
  PropTypes.node,
  PropTypes.arrayOf(PropTypes.node)
]);

export const labelAlignmentShape = PropTypes.oneOf([
  null,
  'left',
  'center',
  'sticky',
  'right'
]);
