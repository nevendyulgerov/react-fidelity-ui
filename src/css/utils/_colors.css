.fidelity-ui-app {
  .color {
    &--primary {
      color: var(--color-primary) !important;
    }

    &--primary-light {
      color: var(--color-primary--light) !important;
    }

    &--primary-dark {
      color: var(--color-primary--dark) !important;
    }

    &--secondary {
      color: var(--color-secondary) !important;
    }

    &--secondary-light {
      color: var(--color-secondary--light) !important;
    }

    &--secondary-dark {
      color: var(--color-secondary--dark) !important;
    }

    &--success {
      color: var(--color-success) !important;
    }

    &--success-light {
      color: var(--color-success--light) !important;
    }

    &--success-dark {
      color: var(--color-success--dark) !important;
    }

    &--warning {
      color: var(--color-warning) !important;
    }

    &--warning-light {
      color: var(--color-warning--light) !important;
    }

    &--warning-dark {
      color: var(--color-warning--dark) !important;
    }

    &--danger {
      color: var(--color-danger) !important;
    }

    &--danger-light {
      color: var(--color-danger--light) !important;
    }

    &--danger-dark {
      color: var(--color-danger--dark) !important;
    }

    &--info {
      color: var(--color-info) !important;
    }

    &--info-light {
      color: var(--color-info--light) !important;
    }

    &--info-dark {
      color: var(--color-info--dark) !important;
    }

    &--body {
      color: var(--color-body) !important;
    }

    &--heading {
      color: var(--color-body--heading) !important;
    }

    &--body-bg {
      color: var(--color-body--bg) !important;
    }

    &--header {
      color: var(--header-color) !important;
    }

    &--header-bg {
      color: var(--header-color-bg) !important;
    }

    &--link {
      color: var(--color-body--link) !important;
    }

    &--facebook {
      color: var(--color-facebook) !important;
    }

    &--google {
      color: var(--color-google) !important;
    }

    &--white {
      color: var(--color-white) !important;
    }

    &--gray-1 {
      color: var(--color-gray-1) !important;
    }

    &--gray-2 {
      color: var(--color-gray-2) !important;
    }

    &--gray-3 {
      color: var(--color-gray-3) !important;
    }

    &--gray-4 {
      color: var(--color-gray-4) !important;
    }

    &--gray-5 {
      color: var(--color-gray-5) !important;
    }

    &--gray-6 {
      color: var(--color-gray-6) !important;
    }

    &--gray-7 {
      color: var(--color-gray-7) !important;
    }

    &--gray-8 {
      color: var(--color-gray-8) !important;
    }

    &--gray-9 {
      color: var(--color-gray-9) !important;
    }

    &--gray-10 {
      color: var(--color-gray-10) !important;
    }

    &--black {
      color: var(--color-black) !important;
    }
  }

  /* bg */
  .bg {
    &--primary {
      background-color: var(--color-primary) !important;
    }

    &--primary-light {
      background-color: var(--color-primary--light) !important;
    }

    &--primary-dark {
      background-color: var(--color-primary--dark) !important;
    }

    &--secondary {
      background-color: var(--color-secondary) !important;
    }

    &--secondary-light {
      background-color: var(--color-secondary--light) !important;
    }

    &--secondary-dark {
      background-color: var(--color-secondary--dark) !important;
    }

    &--success {
      background-color: var(--color-success) !important;
    }

    &--success-light {
      background-color: var(--color-success--light) !important;
    }

    &--success-dark {
      background-color: var(--color-success--dark) !important;
    }

    &--warning {
      background-color: var(--color-warning) !important;
    }

    &--warning-light {
      background-color: var(--color-warning--light) !important;
    }

    &--warning-dark {
      background-color: var(--color-warning--dark) !important;
    }

    &--danger {
      background-color: var(--color-danger) !important;
    }

    &--danger-light {
      background-color: var(--color-danger--light) !important;
    }

    &--danger-dark {
      background-color: var(--color-danger--dark) !important;
    }

    &--info {
      background-color: var(--color-info) !important;
    }

    &--info-light {
      background-color: var(--color-info--light) !important;
    }

    &--info-dark {
      background-color: var(--color-info--dark) !important;
    }

    &--facebook {
      background-color: var(--color-facebook) !important;
    }

    &--google {
      background-color: var(--color-google) !important;
    }

    &--body {
      background-color: var(--color-body) !important;
    }

    &--body--light {
      background-color: var(--color-body--light) !important;
    }

    &--heading {
      background-color: var(--color-body--heading) !important;
    }

    &--body-bg {
      background-color: var(--color-body--bg) !important;
    }

    &--header {
      background-color: var(--header-color) !important;
    }

    &--header-bg {
      background-color: var(--header-color-bg) !important;
    }

    &--link {
      background-color: var(--color-body--link) !important;
    }

    &--white {
      background-color: var(--color-white) !important;
    }

    &--gray-1 {
      background-color: var(--color-gray-1) !important;
    }

    &--gray-2 {
      background-color: var(--color-gray-2) !important;
    }

    &--gray-3 {
      background-color: var(--color-gray-3) !important;
    }

    &--gray-4 {
      background-color: var(--color-gray-4) !important;
    }

    &--gray-5 {
      background-color: var(--color-gray-5) !important;
    }

    &--cgray-6 {
      background-color: var(--color-gray-6) !important;
    }

    &--gray-7 {
      background-color: var(--color-gray-7) !important;
    }

    &--gray-8 {
      background-color: var(--color-gray-8) !important;
    }

    &--gray-9 {
      background-color: var(--color-gray-9) !important;
    }

    &--gray-10 {
      background-color: var(--color-gray-10) !important;
    }

    &--black {
      background-color: var(--color-black) !important;
    }

    &--transparent {
      background: transparent !important;
    }
  }

  /* border-color */
  .border-color {
    &--primary {
      border-color: var(--color-primary) !important;
    }

    &--primary-light {
      border-color: var(--color-primary--light) !important;
    }

    &--primary-dark {
      border-color: var(--color-primary--dark) !important;
    }

    &--secondary {
      border-color: var(--color-secondary) !important;
    }

    &--secondary-light {
      border-color: var(--color-secondary--light) !important;
    }

    &--secondary-dark {
      border-color: var(--color-secondary--dark) !important;
    }

    &--success {
      border-color: var(--color-success) !important;
    }

    &--success-light {
      border-color: var(--color-success--light) !important;
    }

    &--success-dark {
      border-color: var(--color-success--dark) !important;
    }

    &--warning {
      border-color: var(--color-warning) !important;
    }

    &--warning-light {
      border-color: var(--color-warning--light) !important;
    }

    &--warning-dark {
      border-color: var(--color-warning--dark) !important;
    }

    &--danger {
      border-color: var(--color-danger) !important;
    }

    &--danger-light {
      border-color: var(--color-danger--light) !important;
    }

    &--danger-dark {
      border-color: var(--color-danger--dark) !important;
    }

    &--info {
      border-color: var(--color-info) !important;
    }

    &--info-light {
      border-color: var(--color-info--light) !important;
    }

    &--info-dark {
      border-color: var(--color-info--dark) !important;
    }

    &--body {
      border-color: var(--color-body) !important;
    }

    &--body--light {
      border-color: var(--color-body--light) !important;
    }

    &--heading {
      border-color: var(--color-body--heading) !important;
    }

    &--body-bg {
      border-color: var(--color-body--bg) !important;
    }

    &--header {
      border-color: var(--header-color) !important;
    }

    &--header-bg {
      border-color: var(--header-color-bg) !important;
    }

    &--link {
      border-color: var(--color-body--link) !important;
    }

    &--facebook {
      border-color: var(--color-facebook) !important;
    }

    &--google {
      border-color: var(--color-google) !important;
    }

    &--white {
      border-color: var(--color-white) !important;
    }

    &--gray-1 {
      border-color: var(--color-gray-1) !important;
    }

    &--gray-2 {
      border-color: var(--color-gray-2) !important;
    }

    &--gray-3 {
      border-color: var(--color-gray-3) !important;
    }

    &--gray-4 {
      border-color: var(--color-gray-4) !important;
    }

    &--gray-5 {
      border-color: var(--color-gray-5) !important;
    }

    &--gray-6 {
      border-color: var(--color-gray-6) !important;
    }

    &--gray-7 {
      border-color: var(--color-gray-7) !important;
    }

    &--gray-8 {
      border-color: var(--color-gray-8) !important;
    }

    &--gray-9 {
      border-color: var(--color-gray-9) !important;
    }

    &--gray-10 {
      border-color: var(--color-gray-10) !important;
    }

    &--black {
      border-color: var(--color-black) !important;
    }
  }
}
