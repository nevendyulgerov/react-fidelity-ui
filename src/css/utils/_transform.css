.fidelity-ui-app {
  .rotate {
    transition: var(--transition);

    &--90 {
      transform: rotate(90deg) !important;
    }

    &--180 {
      transform: rotate(180deg) !important;
    }

    &--270 {
      transform: rotate(270deg) !important;
    }

    &--360 {
      transform: rotate(360deg) !important;
    }
  }
}
