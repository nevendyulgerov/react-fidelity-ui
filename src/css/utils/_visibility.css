.fidelity-ui-app {
  .visible {
    visibility: visible !important;
  }

  .invisible {
    visibility: hidden !important;
  }
}
