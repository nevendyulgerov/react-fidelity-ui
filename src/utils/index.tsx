/**
 * @description Is null
 * @param val
 */
export const isNull = (val: any) : boolean => val === null;

/**
 * @description Is array
 * @param val
 */
export const isArr = (val: any) : boolean => Array.isArray(val);

/**
 * @description Is object
 * @param val
 */
export const isObj = (val: any) : boolean => typeof val === 'object' && !isArr(val) && !isNull(val);

/**
 * @description Is number
 * @param val
 */
// eslint-disable-next-line no-restricted-globals
export const isNum = (val: any) : boolean => typeof val === 'number' && !isNaN(val);

/**
 * @description Is function
 * @param val
 */
export const isFunc = (val: any) : boolean => typeof val === 'function';

/**
 * @description Is string
 * @param val
 */
export const isStr = (val: any) : boolean => typeof val === 'string';

/**
 * @description Is undefined
 * @param val
 */
export const isUndef = (val: any) : boolean => typeof val === 'undefined';

/**
 * @description Is boolean
 * @param val
 */
export const isBool = (val: any) : boolean => typeof val === 'boolean';

/**
 * @description Is non empty string
 * @param val
 */
export const isNonEmptyStr = (val: any) : boolean => isStr(val) && val !== '';

/**
 * @description Uid
 * @param len
 */
export const uid = (len: number = 7): string => Math.random().toString(35).substr(2, len);

/**
 * @description Get object keys
 * @param obj
 * @returns {Array}
 */
export const getKeys = (obj: any) => Object.keys(obj);

/**
 * @description Each key
 * @param obj
 * @param callback
 */
export const eachKey = (obj: any, callback: (key: string, prop: any, index: number) => void) => {
  getKeys(obj).forEach((key, index) => callback(key, obj[key], index));
};

/**
 * @description Observe intersection
 * @param target
 * @param callback
 */
export const observeIntersection = (target: HTMLElement, callback: (items: []) => void) => {
  if (!isFunc(window.IntersectionObserver)) {
    return console.warn('IntersectionObserver is not supported.');
  }

  // eslint-disable-next-line compat/compat
  const Observe = new IntersectionObserver((items: []) => {
    callback(items);
  });

  Observe.observe(target);

  return Observe;
};

/**
 * @description Get css var
 * @param varName
 */
export const getCssVar = (varName: string): string | null => (
  getComputedStyle(document.documentElement).getPropertyValue(varName)
);

/**
 * @description Set css var
 * @param varName
 * @param varValue
 */
export const setCssVar = (varName: string, varValue: string | null) => {
  document.documentElement.style.setProperty(varName, varValue);
};

/**
 * @description Limit number
 * @param num
 * @param max
 */
export const limitNum = (num: number, max: number) => (num <= max ? num : max);

/**
 * @description Can render prop
 * @param prop
 */
export const isRenderable = (prop: any) => isNonEmptyStr(prop) || isNum(prop) || isObj(prop);
