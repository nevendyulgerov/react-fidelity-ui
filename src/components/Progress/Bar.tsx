import * as React from 'react';
import * as PropTypes from 'prop-types';
import classNames from 'classnames';
import { isNonEmptyStr } from '../../utils';

export type ProgressBarProps = {
  value: number,
  className?: string | null,
  [key: string]: any
};

export interface IProgressBarStyle {
  width: string
}

const ProgressBar = (props: ProgressBarProps) => {
  const { value, className, ...restProps } = props;
  const { style = {} } = restProps;
  const progressInStyle: IProgressBarStyle = {
    ...style,
    width: `${value}%`
  };
  const componentClassName: string = classNames({
    progress__bar: true,
    // @ts-ignore
    [className]: isNonEmptyStr(className)
  });

  return (
    <span
      {...restProps}
      role="progressbar"
      className={componentClassName}
      style={progressInStyle}
    />
  );
};

ProgressBar.propTypes = {
  value: PropTypes.number.isRequired,
  className: PropTypes.string
};

ProgressBar.defaultProps = {
  className: null
};

export default ProgressBar;
