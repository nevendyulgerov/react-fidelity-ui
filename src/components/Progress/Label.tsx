import * as React from 'react';
import * as PropTypes from 'prop-types';
import classNames from 'classnames';
import { labelShape, labelAlignmentShape } from './PropTypes';
import { isNonEmptyStr } from '../../utils';

export type ProgressLabelProps = {
  value: number,
  label?: string | number | React.ReactNode | null,
  alignment?: string | null,
  className?: string | null,
  [key: string]: any
};

export interface IProgressLabelStyle {
  right?: string | undefined
}

const ProgressLabel = (props: ProgressLabelProps) => {
  const { value, label, alignment, className, ...restProps } = props;
  const { style = {} } = restProps;
  const hasAlignment: boolean = isNonEmptyStr(alignment);
  const hasStickyAlignment: boolean = hasAlignment && alignment === 'sticky';
  const progressLabelStyle: IProgressLabelStyle = hasStickyAlignment
    ? {
      ...style,
      right: `${100 - value}%`
    }
    : style;

  const componentClassName: string = classNames({
    progress__label: true,
    [`progress__label--${alignment}`]: hasAlignment,
    // @ts-ignore
    [className]: isNonEmptyStr(className)
  });

  return (
    <div
      {...restProps}
      className={componentClassName}
      style={progressLabelStyle}
    >
      {label}
    </div>
  );
};

ProgressLabel.propTypes = {
  value: PropTypes.number,
  label: labelShape.isRequired,
  alignment: labelAlignmentShape,
  className: PropTypes.string
};

ProgressLabel.defaultProps = {
  value: 0,
  alignment: null,
  className: null
};

export default ProgressLabel;
