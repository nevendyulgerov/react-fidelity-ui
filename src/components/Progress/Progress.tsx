import * as React from 'react';
import * as PropTypes from 'prop-types';
import classNames from 'classnames';
import ProgressBar from './Bar';
import ProgressLabel from './Label';
import { typeShape, sizeShape, labelShape, labelAlignmentShape } from './PropTypes';
import { isNonEmptyStr, limitNum, isRenderable } from '../../utils';

export type ProgressProps = {
  type?: string,
  size?: string | null,
  value: number,
  label?: string | number | React.ReactNode | null,
  labelAlignment?: string | null,
  className?: string | null,
  rounded?: boolean,
  elevated?: boolean,
  hovered?: boolean,
  [key: string]: any
};

const Progress = (props: ProgressProps) => {
  const { type, size, value, label, labelAlignment, className, rounded, elevated, hovered, ...restProps } = props;
  const maxValue: number = 100;
  const formattedValue: number = limitNum(value, maxValue);
  const hasLabelSlot: boolean = isRenderable(label);
  const componentClassName: string = classNames({
    progress: true,
    'progress--rounded': rounded,
    'progress--elevated': elevated,
    'progress--hovered': hovered,
    [`progress--${type}`]: isNonEmptyStr(type),
    [`progress--${size}`]: isNonEmptyStr(size),
    // @ts-ignore
    [className]: isNonEmptyStr(className)
  });

  return (
    <div
      {...restProps}
      className={componentClassName}
    >
      <ProgressBar
        role="progressbar"
        aria-valuenow={formattedValue}
        aria-valuemin={0}
        aria-valuemax={maxValue}
        value={formattedValue}
      />

      {hasLabelSlot && (
        <ProgressLabel
          label={label}
          value={formattedValue}
          alignment={labelAlignment}
        />
      )}
    </div>
  );
};

Progress.propTypes = {
  type: typeShape,
  size: sizeShape,
  value: PropTypes.number,
  label: labelShape,
  labelAlignment: labelAlignmentShape,
  className: PropTypes.string,
  rounded: PropTypes.bool,
  elevated: PropTypes.bool,
  hovered: PropTypes.bool
};

Progress.defaultProps = {
  type: 'secondary',
  size: null,
  value: 0,
  label: null,
  labelAlignment: null,
  className: null,
  rounded: true,
  elevated: false,
  hovered: false
};

export default Progress;
