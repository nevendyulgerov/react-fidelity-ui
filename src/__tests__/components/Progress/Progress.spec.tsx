import * as React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Progress from '../../../components/Progress';
import { isObj } from '../../../utils';

describe('Components / Progress', () => {
  it('matches snapshot', () => {
    const { baseElement } = render(<Progress value={25} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('displays bar node', () => {
    const { baseElement } = render(<Progress value={25} />);
    const progressLabelNode = baseElement.querySelector('.progress__bar');

    expect(isObj(progressLabelNode)).toBe(true);
  });

  it('displays label node when label is a non empty string', () => {
    const { baseElement } = render(
      <Progress
        value={25}
        label="lorem ipsum dolor sit amet"
      />
    );
    const progressLabelNode = baseElement.querySelector('.progress__label');

    expect(isObj(progressLabelNode)).toBe(true);
  });

  it('does not display label node when label is an empty string', () => {
    const { baseElement } = render(
      <Progress
        value={25}
        label=""
      />
    );
    const progressLabelNode = baseElement.querySelector('.progress__label');

    expect(isObj(progressLabelNode)).toBe(false);
  });

  it('displays label node when label is number', () => {
    const { baseElement } = render(
      <Progress
        value={25}
        label={25}
      />
    );
    const progressLabelNode = baseElement.querySelector('.progress__label');

    expect(isObj(progressLabelNode)).toBe(true);
  });

  it('displays label node when label is node', () => {
    const { baseElement } = render(
      <Progress
        value={25}
        label={<span>Label</span>}
      />
    );
    const progressLabelNode = baseElement.querySelector('.progress__label');

    expect(isObj(progressLabelNode)).toBe(true);
  });

  it('attaches className "progress" to wrapper', () => {
    const { baseElement } = render(<Progress value={25} />);
    const progressNode = baseElement.querySelector('.progress');

    expect(isObj(progressNode)).toBe(true);
  });

  it('attaches className "progress--rounded" to wrapper when "rounded" prop is set to "true"', () => {
    const { baseElement } = render(<Progress value={25} rounded />);
    const progressNode = baseElement.querySelector('.progress');

    // @ts-ignore
    expect(progressNode.classList.contains('progress--rounded')).toBe(true);
  });

  it('attaches className "progress--elevated" to wrapper when "elevated" prop is set to "true"', () => {
    const { baseElement } = render(<Progress value={25} elevated />);
    const progressNode = baseElement.querySelector('.progress');

    // @ts-ignore
    expect(progressNode.classList.contains('progress--elevated')).toBe(true);
  });

  it('attaches className "progress--hovered" hovered wrapper when "hovered" prop is set to "true"', () => {
    const { baseElement } = render(<Progress value={25} hovered />);
    const progressNode = baseElement.querySelector('.progress');

    // @ts-ignore
    expect(progressNode.classList.contains('progress--hovered')).toBe(true);
  });

  it('attaches className "progress--{type}" matching "type" prop', () => {
    let type = 'primary';
    const { baseElement, rerender } = render(<Progress value={25} type={type} />);
    const progressNode = baseElement.querySelector('.progress');

    // @ts-ignore
    expect(progressNode.classList.contains(`progress--${type}`)).toBe(true);

    type = 'secondary';
    rerender(<Progress value={25} type={type} />);

    // @ts-ignore
    expect(progressNode.classList.contains(`progress--${type}`)).toBe(true);

    type = 'info';
    rerender(<Progress value={25} type={type} />);

    // @ts-ignore
    expect(progressNode.classList.contains(`progress--${type}`)).toBe(true);

    type = 'warning';
    rerender(<Progress value={25} type={type} />);

    // @ts-ignore
    expect(progressNode.classList.contains(`progress--${type}`)).toBe(true);

    type = 'success';
    rerender(<Progress value={25} type={type} />);

    // @ts-ignore
    expect(progressNode.classList.contains(`progress--${type}`)).toBe(true);

    type = 'danger';
    rerender(<Progress value={25} type={type} />);

    // @ts-ignore
    expect(progressNode.classList.contains(`progress--${type}`)).toBe(true);
  });

  it('attaches className "progress--{size}" matching "size" prop', () => {
    let size = 'xs';
    const { baseElement, rerender } = render(<Progress value={25} size={size} />);
    const progressNode = baseElement.querySelector('.progress');

    // @ts-ignore
    expect(progressNode.classList.contains(`progress--${size}`)).toBe(true);

    size = 'sm';
    rerender(<Progress value={25} size={size} />);

    // @ts-ignore
    expect(progressNode.classList.contains(`progress--${size}`)).toBe(true);

    size = 'md';
    rerender(<Progress value={25} size={size} />);

    // @ts-ignore
    expect(progressNode.classList.contains(`progress--${size}`)).toBe(true);

    size = 'lg';
    rerender(<Progress value={25} size={size} />);

    // @ts-ignore
    expect(progressNode.classList.contains(`progress--${size}`)).toBe(true);

    size = 'xl';
    rerender(<Progress value={25} size={size} />);

    // @ts-ignore
    expect(progressNode.classList.contains(`progress--${size}`)).toBe(true);
  });

  it('attaches className matching "className" prop', () => {
    const className = 'my-selector';
    const { baseElement } = render(<Progress value={25} className={className} />);
    const progressNode = baseElement.querySelector('.progress');

    // @ts-ignore
    expect(progressNode.classList.contains(className)).toBe(true);
  });

  it('adds any prop to wrapper', () => {
    // test style prop
    const style = { height: '20px' };
    const { baseElement, rerender } = render(<Progress value={25} style={style} />);
    const progressNode = baseElement.querySelector('.progress');

    // @ts-ignore
    expect(getComputedStyle(progressNode).height).toBe(style.height);

    // test event/callback prop
    let isClickTriggered = false;
    const onClick = () => {
      isClickTriggered = true;
    };
    rerender(<Progress value={25} onClick={onClick} />);

    // @ts-ignore
    fireEvent.click(progressNode);

    expect(isClickTriggered).toBe(true);

    // test attribute prop
    const role = 'button';
    rerender(<Progress value={25} role={role} />);

    // @ts-ignore
    expect(progressNode.getAttribute('role')).toBe(role);
  });
});
