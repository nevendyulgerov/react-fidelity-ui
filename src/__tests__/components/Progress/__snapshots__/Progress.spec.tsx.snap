// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components / Progress matches snapshot 1`] = `
<body>
  <div>
    <div
      class="progress progress--rounded progress--secondary"
    >
      <span
        class="progress__bar"
        style="width: 25%;"
      />
    </div>
  </div>
</body>
`;
